src/mmda/predictors/heuristic_predictors/dictionary_word_predictor.py

```diff
@@ -1,27 +1,76 @@
 """
-DictionaryWordPredictor -- Reads rows of text into dehyphenated words.
+DictionaryWordPredictor -- Reads `tokens` and converts them into whole `words`.
+
+Let's consider 4 consecutive rows:
+
+    This is a few-shot learn-
+    ing paper. We try few-
+    shot techniques. These
+    methods are useful.
+
+This predictor tries to:
+    1. merge tokens ["learn", "-", "ing"] into "learning"
+    2. merge tokens ["few", "-", "shot"] into "few-shot"
+    3. keep tokens ["These", "methods"] separate
+    4. keep tokens ["useful", "."] separate
+
+This technique requires 2 passes through the data:
+
+    1. Build a dictionary of valid words, e.g. if "learning" was ever used, then we can merge
+       ["learn", "-", "ing"].
+
+    2. Classify every pair of tokens as belonging to the *same* or *different* words.
+
+@kylel, @rauthur
 
-@rauthur
 """
 
-import string
-from typing import Optional, Set, List
+import os
+from typing import Optional, Set, List, Tuple, Iterable, Dict
 
+from collections import defaultdict
+from mmda.parsers import PDFPlumberParser
 from mmda.predictors.base_predictors.base_predictor import BasePredictor
-from mmda.types.metadata import Metadata
-from mmda.types.annotation import Annotation, Span, SpanGroup
-from mmda.types.document import Document
+from mmda.predictors import WhitespacePredictor
```

Comment:
I think this needs to be `from mmda.predictors.heuristic_predictors import WhitespacePredictor`, right?